library/src/main/java/com/devbrackets/android/playlistcore/service/PlaylistServiceCore.java

```diff
@@ -738,6 +738,20 @@ protected void performPlay() {
                 videoPlayer.play();
             }
         }
+        
+        //Immediately pauses the media
```

Comment:
@brianwernick I agree with you, ins't a good approach moving these methods in `performPlay()`, but the PlaylistServiceCore class will need some re-work, because the delay problem come with the methods called right after calling the `play()` from `performPlay()`, I've tracked the time running these methods with my device:

mediaProgressPoll.start() took 1ms
setPlaybackState(PlaybackState.PLAYING) took 7ms
setupForeground() took 8ms
requestAudioFocus() took 28ms
updateNotification() took 84ms
updateRemoteViews() took 112ms

Total of 240ms, here the audio already played a bit before stopping it after the method finished.